waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -187,37 +192,78 @@ protected void processCmd(CommandLine cmd) throws SubCommandFailedException {
+                    buildServersZKPartitionAssignmentsConsistencyValidation(
+                        rpcClient, clusterManager, partitionsValidationResultList
+                    );
+
+                // Step3: Build Server-Storage connectivity validations
+                CompletableFuture<Void> connectivityValidationFuture =
+                    buildServerStorageConnectivityValidation(
+                        rpcClient,
+                        clusterManager,
+                        storeMetadata.getReplicaAssignments(),
+                        partitionsValidationResultList
+                    );
+
+                // Verify validations for Step2 and Step3
+                try {
+                    consistencyValidationFuture.get(TIMEOUT_IN_SECONDS, TimeUnit.SECONDS);
+                } catch (ExecutionException | TimeoutException exception) {
+                    failMissingValidationResults(
+                        partitionsValidationResultList,
+                        ValidationResult.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY,
+                        exception.toString()
+                    );
+                }
 
-                for (PartitionValidationResults results : partitionsValidationResultList) {
-                    if (!results.validationResultsMap.containsKey(ValidationResults.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY)) {
-                        ValidationResults validationResults = new ValidationResults(ValidationResults.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY,
-                                ValidationResults.Status.FAILURE, "Timeout exception");
-                        results.validationResultsMap.put(ValidationResults.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY, validationResults);
-                    }
+                try {
+                    connectivityValidationFuture.get(TIMEOUT_IN_SECONDS, TimeUnit.SECONDS);
+                } catch (ExecutionException | TimeoutException exception) {
+                    failMissingValidationResults(
+                        partitionsValidationResultList,
+                        ValidationResult.ValidationType.SERVER_STORAGE_CONNECTIVITY,
+                        exception.toString()
+                    );
                 }
-                verifyValidation(ValidationResults.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY, partitionsValidationResultList);
 
-            } catch (Exception e) {
+                verifyValidation(
+                    ValidationResult.ValidationType.PARTITION_ASSIGNMENT_ZK_SERVER_CONSISTENCY,
+                    partitionsValidationResultList
+                );
+
+                verifyValidation(
+                    ValidationResult.ValidationType.SERVER_STORAGE_CONNECTIVITY,
+                    partitionsValidationResultList
+                );
+            } catch (RuntimeException e) {
```

Comment:
isn't this exception also covered in `catch (Exception e)` in the else case? do we need this (assuming we don't want to print any specific error corresponding to it.) ??

Reply:
yes it is. But, I was getting a findbugs warning that a generic Exception is caught :(

Reply:
ok ok, got it. let it be then.